src/systemctl/systemctl-edit.c

```diff
@@ -309,6 +309,49 @@ static int unit_file_create_copy(
         return 0;
 }
 
+static void exec_editor(const char *editor, char **paths, size_t argc, char ***args_save) {
+        char **editor_args, **args, **original_path, **tmp_path;
+        size_t n_editor_args, i = 1;
+
+        args = *args_save;
+
+        if (isempty(editor))
+                return;
+
+        editor_args = strv_split(editor, WHITESPACE);
```

Comment:
you leak this memory. even if we'll shortly exec some other process here, and the leak doesn't really matter too much, as a matter of hygiene, please don't leak memory